### PR TITLE
Cli updates

### PR DESCRIPTION
**Pull request by bwagner5**
*Issue #, if available:*
N/A

*Description of changes:*
Enhance the CLI pkg w/ 
 - StringSliceFlag (accepts a list of strings from the command line)
 - StringOptionsFlag (accepts a string from the command line and performs validation based on options based in when the flag is registered)
 - Expose raw cobra command for advanced usage
 - Expose CLI processing logic for pluggable usage
 - Added test coverage

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 	"github.com/aws/amazon-ec2-instance-selector/pkg/selector"
 	"github.com/aws/amazon-ec2-instance-selector/pkg/selector/outputs"
 	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/spf13/cobra"
```
[the remark this answers is not in the document]
Comment (bwagner5) [reply cmd/main.go:27]: no, it's already used in this project, just depending on it in another file.

File: pkg/cli/cli.go [hunk 2]
```diff
@@ -182,8 +188,8 @@ func (cl *CommandLineInterface) setUntouchedFlagValuesToNil() error {
 	return nil
 }
 
-// processRangeFilterFlags sets min and max to the appropriate 0 or maxInt bounds based on the 3-tuple that a user specifies for base flag, min, and/or max
-func (cl *CommandLineInterface) processRangeFilterFlags() error {
+// ProcessRangeFilterFlags sets min and max to the appropriate 0 or maxInt bounds based on the 3-tuple that a user specifies for base flag, min, and/or max
+func (cl *CommandLineInterface) ProcessRangeFilterFlags() error {
```
Comment (haugenj) [inline pkg/cli/cli.go:192]: why do we need to make this and SetUntouchedFlagValuesToNil public? It doesn't look like you're using them in any new locations

File: pkg/cli/types.go [hunk 3]
```diff
@@ -51,7 +51,7 @@ type validator = func(val interface{}) error
 
 // CommandLineInterface is a type to group CLI funcs and state
 type CommandLineInterface struct {
-	rootCmd       *cobra.Command
+	Command       *cobra.Command
```
Comment (haugenj) [inline pkg/cli/types.go:54]: Why change to public?

Comment (bwagner5) [reply pkg/cli/types.go:54]: so that this command can be used in a larger cobra app

File: pkg/cli/cli.go (hunk 2 above)
Comment (bwagner5) [reply pkg/cli/cli.go:192]: the cli package can be used outside of this project to perform integrations with instance-selector and still have access to instance-selector helpers for special flags.  Instance-selector is able to call ParseAndValidate which performs all processing, but when used in another cobra app, we need the flexibility to call individual processing functions.

File: pkg/cli/flags.go [hunk 4]
```diff
@@ -47,49 +47,90 @@ func (cl *CommandLineInterface) RatioFlag(name string, shorthand *string, defaul
...
+	cl.StringFlagOnFlagSet(cl.suiteFlags, name, shorthand, defaultValue, description, validationFn)
+}
+
+// SuiteStringOptionsFlag creates and registers a flag accepting a string and valid options for use in validation.
+// Suite flags will be grouped in the middle of the output --help
+func (cl *CommandLineInterface) SuiteStringOptionsFlag(name string, shorthand *string, defaultValue *string, description string, validOpts []string) {
+	cl.StringOptionsFlagOnFlagSet(cl.suiteFlags, name, shorthand, defaultValue, description, validOpts)
+}
+
+// SuiteStringSliceFlag creates and registers a flag accepting a string and valid options for use in validation.
```
Comment (brycahta) [inline pkg/cli/flags.go:128]: nit: accepting a **string slice** and valid options

**Comment by bwagner5** [comment]
`removeIntersectingArgs` won't affect other command's parsing logic and cannot be used in a larger cobra app since the parse logic will be the responsibility of the rootCmd of that app. This function is only available for the actual ec2-instance-selector CLI driver.

**Pull request merged**